superset/assets/javascripts/explore/components/controls/AnnotationLayer.jsx

```diff
@@ -0,0 +1,558 @@
+    }
+  }
+
+  submitAnnotation() {
+    this.applyAnnotation();
+    this.props.close();
+  }
+
+  renderValueConfiguration() {
+    const { annotationType, value, valueOptions, isLoadingOptions } = this.state;
+    let label = '';
+    let description = '';
+    if (annotationType === AnnotationTypes.NATIVE) {
+      label = 'Annotation Layer';
+      description = 'Select the Annotation Layer you would like to use.';
+    } else if (isQueryAnnotationType(annotationType)) {
+      label = 'Slice';
+      description = 'Use a pre defined Superset Slice as a source for annotations and overlays. ' +
+        'your Slice must be one of these visualization types: ' +
+        `[${supportedSliceTypes(annotationType).map(x => vizTypes[x].label).join(', ')}]`;
+    } else if (annotationType === AnnotationTypes.FORMULA) {
+      label = 'Formula';
+      description = 'Expects a formula with depending time parameter `x`' +
+        ' in milliseconds since epoch. mathjs is used to evaluate the formulas.';
+    }
+    if (isQueryAnnotationType(annotationType) || annotationType === AnnotationTypes.NATIVE) {
+      return (
+        <SelectControl
+          name="annotation-layer-value"
+          showHeader
+          hovered
+          description={description}
+          label={label}
+          placeholder=""
+          options={valueOptions}
+          isLoading={isLoadingOptions}
+          value={value}
+          onChange={this.handleValue}
+          validationErrors={!value ? ['Mandatory'] : []}
+        />
+      );
+    } if (annotationType === AnnotationTypes.FORMULA) {
+      return (
+        <TextControl
+          name="annotation-layer-value"
+          hovered
+          showHeader
+          description={description}
+          label={label}
+          placeholder=""
+          value={value}
+          onChange={this.handleValue}
+          validationErrors={this.validateFormula(value, annotationType) ? ['Bad formula.'] : []}
+        />
+      );
+    }
+    return '';
+  }
+
+  renderSliceConfiguration() {
+    const { annotationType, value, valueOptions, overrides, titleColumn,
+      timeColumn, intervalEndColumn, descriptionColumns } = this.state;
+    const slice = (valueOptions.find(x => x.value === value) || {}).slice;
+    if (isQueryAnnotationType(annotationType) && slice) {
+      const columns = (slice.form_data.groupby || []).concat(
+        (slice.form_data.all_columns || [])).map(x => ({ value: x, label: x }));
+      const timeColumnOptions = slice.form_data.include_time ?
+        [{ value: '__timestamp', label: '__timestamp' }].concat(columns) : columns;
+      return (
+        <div style={{ marginRight: '2rem' }}>
+          <PopoverSection
+            isSelected
+            onSelect={() => {
+            }}
+            title="Annotation Slice Configuration"
+            info={
+              'This section allows you to configure how to use the slice ' +
```

Comment:
you can use back ticks \`\` to avoid concat, it handles spaces fine with newlines